lib/dolly/property.rb

```diff
@@ -12,19 +12,42 @@ def initialize opts = {}
       @default = opts.delete(:default)
       @default = @default.clone if @default && CANT_CLONE.none? { |klass| @default.is_a? klass }
       @value = @default if @default
+      @subproperties = {}
       warn 'There are some unprocessed options!' if opts.present?
     end
 
     def value
```

Comment:
in general it seems like this method is now doing a lot of work and wondering if it might be worth splitting up